### ISSUE TITLE
Provide the ability to specify a custom sort algorithm for events that show on the exact same time slot

### ISSUE THREAD

**Issue by darkrift** [intljusticemission/react-big-calendar#838]
I have events that I want to sort by title when they are due at the exact same time.

Currently the sort is exclusively made from the startAccessor which isn't always sufficient.

The feature request is to be able to provide to the calendar object a new props to specify an additional sort function in case events would result in the same time slot/length.

I'm currently defining my own components with a single line modification for the agenda, and I'll have to do the same with the month view which calls "sortEvents" as a quick fix.

**Comment by arecvlohe** [comment]
Open to accepting PRs on this but your specific issue may be fixed in #677

**Comment by jehartzog** [comment]
In case anyone else comes across this, if you really want to do this for your project you can fork this repo and run a custom version.

Here is the custom sorting [we used](https://github.com/jehartzog/react-big-calendar/blob/8ceaad46b6b9fc6e17c66ac20fc760c28d61fe8e/src/utils/eventLevels.js#L98):

`/src/utils/eventLevels.js`
```js
  return (
    startSort || // sort by start Day first
    Math.max(durB, 1) - Math.max(durA, 1) || // events spanning multiple days go first
    !!accessors.allDay(evtB) - !!accessors.allDay(evtA) || // then allDay single day events
    evtA.eventType.localeCompare(evtB.eventType) || // Custom sort by eventType
    +accessors.start(evtA) - +accessors.start(evtB) || // then sort by start time
    evtA.title.localeCompare(evtB.title) // Custom sort by title
  )
```

If you want to run this quick and easy from your own forked github repo, remember you'll need to un-gitignore `/lib`, and make sure to run `npm run build` before you push to your fork branch.

Also be sure to branch from a tagged release commit version, we accidentally used master at first and it completely failed.
